### ISSUE TITLE
Linking of SYCL examples fails with undefined symbols in case of one-CPU build

### ISSUE THREAD

**Issue by rlnx** [intel/daal#198]
Linking of SYCL examples fails with undefined symbols in case of one-CPU build. 

**To Reproduce**
1. Build library with one CPU:
   ```sh
   make -j24 _daal _release_c CORE.ALGORITHMS.CUSTOM="pca" REQDBG=yes REQCPU=avx2
   ```
2. Try to build SYCL example:
   ```sh
   make sointel64 mode=build example=pca_cor_dense_batch
   ```
4. See undefined references to MKL LAPACK functions, e.g., `fpk_lapack_avx_spotrf`

**Examples build log**
```
clang++ -Wall -w -I./source/utils -fsycl -DONEAPI_DAAL_USE_MKL_GPU_FUNC ./source/data_management/numeric_table_usm.cpp -o _results/clang_intel64_parallel_so/numeric_table_usm.exe -Wl,--start-group "latest/lib/intel64"/libdaal_core.so "latest/lib/intel64"/libdaal_thread.so -lpthread -ldl -lOpenCL -L"/export/users/cache/tbb/lnx/20
21.1-beta02-fix/tbb/latest/lib/intel64/gcc4.8" -ltbb -ltbbmalloc -foffload-static-lib="latest/lib/intel64"/libdaal_sycl.a -Wl,--end-group
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx_spotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_mic_spotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_ssse3_dpotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_mic_dpotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_dpotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_spotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_sse42_dpotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx_dpotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_ssse3_spotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_ssse3_dpotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_spotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_mic_dpotrf'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_dpotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx_spotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_ssse3_spotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_sse42_spotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx_dpotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_avx512_mic_spotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_sse42_dpotrs'
latest/lib/intel64/libdaal_core.so: undefined reference to `fpk_lapack_sse42_spotrf'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Environment:**
 - OS: Ubuntu 18.04
 - Compiler: Intel(R) oneAPI DPC++ Compiler 2021.1-beta03 (2019.10.0.1029)
 - Version: Current master

**Comment by rlnx** [comment]
@prutskov, it looks like there is problem with the following code in `lapack_gpu.cpp`, it does not check for allowed CPUs 
```cpp
#define CALL_LAPACK_CPU(fptype, cpuId, funcName, ...) \
        switch (cpuId) \
    {                                                                                                                   \
    case daal::CpuType::ssse3: Lapack<fptype, daal::CpuType::ssse3>::funcName(__VA_ARGS__); break;                      \
    case daal::CpuType::sse42: Lapack<fptype, daal::CpuType::sse42>::funcName(__VA_ARGS__); break;                      \
    case daal::CpuType::avx:   Lapack<fptype, daal::CpuType::avx>::funcName(__VA_ARGS__); break;                        \
    case daal::CpuType::avx2:  Lapack<fptype, daal::CpuType::avx2>::funcName(__VA_ARGS__); break;                       \
    case daal::CpuType::avx512_mic: Lapack<fptype, daal::CpuType::avx512_mic>::funcName(__VA_ARGS__); break;            \
    case daal::CpuType::avx512:  Lapack<fptype, daal::CpuType::avx512>::funcName(__VA_ARGS__); break;                   \
    case daal::CpuType::avx512_mic_e1: Lapack<fptype, daal::CpuType::avx512_mic_e1>::funcName(__VA_ARGS__); break;      \
    default: Lapack<fptype, daal::CpuType::sse2>::funcName(__VA_ARGS__); break;                                         \
    }
```

**Comment by prutskov** [comment]
Fix prepared in PR [#199](https://github.com/intel/daal/pull/199)
